TestTools/StreamChatTestMockServer/MockServer/ChannelResponses.swift

```diff
@@ -338,19 +339,29 @@ public extension StreamMockServer {
             
             if messagesCount > 0 {
                 for messageIndex in 1...messagesCount {
-                    let timeInterval = TimeInterval(messageIndex * 1000 - 123_456_789)
-                    let timestamp = TestData.stringTimestamp(Date(timeIntervalSinceNow: timeInterval))
-                    let message = mockMessage(
-                        TestData.toJson(.message)[JSONKey.message] as? [String : Any],
-                        channelId: channelDetails?[channelKey.id.rawValue] as? String,
-                        messageId: TestData.uniqueId,
-                        text: String(messageIndex),
-                        user: author,
-                        createdAt: timestamp,
-                        updatedAt: timestamp
+                    let channelId = channelDetails?[channelKey.id.rawValue] as? String
+                    let messageId = TestData.uniqueId
+                    let newMessage = generateMessage(
+                        withIndex: messageIndex,
+                        withId: messageId,
+                        channelId: channelId,
+                        author: author,
+                        replyCount: replyCount
                     )
-                    messages.append(message)
-                    saveMessage(message)
+                    messages.append(newMessage)
+                    
+                    if replyCount > 0 {
+                        for replyIndex in 1...replyCount {
+                            generateMessage(
+                                withIndex: replyIndex,
+                                withId: TestData.uniqueId,
+                                inThread: true,
```

Comment:
Instead of inThread: true, why don't we use the parentId, wouldn't that simplify the `saveReply` function? 
```suggestion
                                parentId: newMessage.id,
```

Reply:
Nice catch, done 👍